### PR TITLE
Initial validation highlighting

### PR DESCRIPTION
**Pull request by HEdingfield**
* Adds validation highlighting to the GUI when clicking the "add" buttons for candidates and CVRs.
* Changes `ContestConfig.validate()` and associated methods to return a set of validation errors instead of an `isValid` boolean.
* Fixes bug where `treatBlankAsUndeclaredWriteIn` validation failure for certain providers wouldn't actually fail validation.
* Moves validation of provided overvote delimiter and overvote label into `performBasicCvrSourceValidation` so user is alerted when clicking the "add" button in the CVR tab.

First pass as outlined in https://github.com/BrightSpots/rcv/issues/137#issuecomment-1214427620.

### CODE REVIEW

File: src/main/java/network/brightspots/rcv/Main.java [hunk 1]
```diff
@@ -27,7 +27,7 @@
 public class Main extends GuiApplication {
 
   public static final String APP_NAME = "RCTab";
-  public static final String APP_VERSION = "1.2.0";
+  public static final String APP_VERSION = "1.3.0";
```
Comment (tarheel) [inline src/main/java/network/brightspots/rcv/Main.java:30]: should maybe be in a separate PR?

**Review by tarheel — COMMENTED** [review]
This is great! Could you post a screenshot or two?

**Review by HEdingfield — COMMENTED** [review]
Great review, thank you!

**Comment by HEdingfield** [comment]
Screenshot here:
Capture

File: src/main/java/network/brightspots/rcv/ContestConfig.java [hunk 2]
```diff
@@ -703,45 +701,52 @@ && getMaxRankingsAllowed() < MIN_MAX_RANKINGS_ALLOWED)) {
...
         if (!isMultiSeatBottomsUpWithThresholdEnabled()) {
-          isValid = false;
+          validationErrors.add(
+              ValidationError
+                  .RULES_NUMBER_OF_WINNERS_WINNER_ELECTION_MODE_PERCENTAGE_THRESHOLD_DISAGREEMENT);
           Logger.severe(
               "If numberOfWinners is zero, winnerElectionMode must be \"%s\" and "
                   + "multiSeatBottomsUpPercentageThreshold must be specified!",
               WinnerElectionMode.MULTI_SEAT_BOTTOMS_UP_USING_PERCENTAGE_THRESHOLD);
         } else if (getMultiSeatBottomsUpPercentageThreshold() == null) {
```
[the remark this answers is not in the document]
Comment (tarheel) [reply src/main/java/network/brightspots/rcv/ContestConfig.java:747]: If numWinners is zero, two things must be true:
- `isMultiSeatBottomsUpWithThresholdEnabled()`
- `getMultiSeatBottomsUpPercentageThreshold !== null`

So I would just check each of these conditions independently here. It's a little confusing because the chain of requirements is:
zero winners --> multi-seat bottoms-up --> non-null threshold

But I would phrase the first check as "zero winners --> must use multi-seat bottoms-up" and the second check as "zero winners --> must set threshold."

This is a very minor thing, though, so do what you think is best.

**Review by tarheel — APPROVED** [review]

File: src/main/java/network/brightspots/rcv/ContestConfig.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (HEdingfield) [reply src/main/java/network/brightspots/rcv/ContestConfig.java]: That makes sense, thanks! Changed it.
